### PR TITLE
[docs] First draft for updated `README.md`

### PR DESCRIPTION
**Pull request by pablo-gar**
**Issue and/or context:**
Adresses #1021 and #1020 

It adds to the readme the following:

- Small editorial changes to make it more descriptive of TileDB-SOMA capabilities.
- Getting started links
- A section "What can TileDB-SOMA do?" explaining value props of the product
- A "Quick start" section with documentation links, brief explanation of SOMA objects and usage examples for python (R to be completed later)

**Notes for Reviewer:**

- Please review redaction, technical accuracy, and contents.
- The quick start sections for the APIs were moved to the wiki per @johnkerl documentation. ~~I believe the [Python quick start](https://github.com/single-cell-data/TileDB-SOMA/tree/pablo-gar/readme_update#python-quick-start) section may be too long for this README. Please suggest any of the following:~~
   - ~~How to shorten it~~
   - ~~Whether to strip it or move it somewhere else.~~
   - ~~Consider there may an R equivalent.~~

**Extra notes**
The text contains a few placeholder links for the doc-site,  and since the doc-site is still not up we may not want to publish these changes. Please comment on that as well.

### CODE REVIEW

**Comment by pablo-gar** [comment]
@johnkerl please feel free to add any other reviewers that you may deem necessary

File: README.md [hunk 1]
```diff
@@ -7,43 +7,84 @@
...
+Intended to be used for single-cell data, TileDB-SOMA provides Python and R APIs to allow for storage and data access patterns at scale and for larger-than-memory operations:
+
+* Create and write large volumes of data.
+* Open and read data at low latency, locally and from the cloud.
+* Query and access interconnected arrays efficiently and at low latency.
+
+TiledDB-SOMA provides interoperability with existing single-cell toolkits:
+
+* Load and create [anndata](https://anndata.readthedocs.io/en/latest/) objects.
+* Load and create [Seurat](https://anndata.readthedocs.io/en/latest/) objects. *Coming soon*.
```
Comment (aaronwolen) [inline README.md:33]: ``
```suggestion
* Load and create [Seurat](https://satijalab.org/seurat/) objects. *Coming soon*.
```

File: README.md [hunk 2]
```diff
@@ -7,43 +7,84 @@
...
+
+TiledDB-SOMA provides interoperability with existing Python or R data structures:
+
+* From Python create PyArrow objects, SciPy sparse matrices, NumPy arrays, and pandas data frames.
+* From R create RArrow objects, Matrix sparse matrices, and R-base dense matrices.
```
Comment (aaronwolen) [inline README.md:38]:
```suggestion
* From R create R Arrow objects, sparse matrices (via the [Matrix](https://cran.r-project.org/package=Matrix) package), and standard data frames and (dense) matrices.
```

File: README.md [hunk 3]
```diff
@@ -7,43 +7,84 @@
...
+
+The capabilities of TileDB-SOMA lay on the different read, access, and query patterns that each of the main implementations of SOMA objects provide:
+
+* `DenseNDArray` a dense, N-dimensional array, with offset (zero-based) integer indexing on each dimension.
+* `SparseNDArray` same as `DenseNDArray` but sparse, and it supports point indexing (disjoint index access)
+* `DataFrame`  a multi-column table with a user-defined columns names and value types, with support for point indexing. 
+* `Collection` a persistent container of named SOMA objects.
+* `Experiment` is a class that represents a single-cell experiment. It always contains two objects:
+	* `obs`: a  `DataFrame` with primary annotations on the observation axis.
+	* `ms`: a  `Collection` of measurements, each composed of `X` matrices and axis annotation matrices or data frames.
```
Comment (aaronwolen) [inline README.md:66]: Let's add periods to all bullets or none

**Review by aaronwolen — APPROVED** [review]
Very nice! I made a couple of suggestions but this is a nice improvement.

**Review by thetorpedodog — COMMENTED** [review]
I have no comment on the contents themselves, just some formatting and link issues.

It might also be worth running through prettier to ensure everything is formatted uniformly, like in https://github.com/single-cell-data/SOMA/pull/134.

File: README.md [hunk 4]
```diff
@@ -7,43 +7,84 @@
...
 The TileDB-SOMA package is a C++ library with APIs in Python and R, using [TileDB
 Embedded](https://github.com/TileDB-Inc/TileDB) to implement the
 [SOMA specification](https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md).
 
-# Community
+Get started on using TileDB-SOMA:
+
+* [Quick start](#quick-start).
+* Python [documentation](#tiledb-soma) | [tutorials](#tiledb-soma). *Under development*.
+* R [documentation](#tiledb-soma) | [tutorials](#tiledb-soma). *Under development*.
```
[the remark this answers is not in the document]
Comment (ebezzi) [reply README.md:20]: I think they're supposed to link to the docpages that do not exist yet.

File: README.md [hunk 5]
```diff
@@ -7,43 +7,84 @@
...
+
+The TileDB-SOMA doc-site (under development), contains the reference documentation and tutorials.
+
+Reference documentation can also be accessed directly from Python `help(tiledsoma)` or R `help(package = "tiledbsoma")`.
+
+### Main SOMA Objects
+
+The capabilities of TileDB-SOMA lay on the different read, access, and query patterns that each of the main implementations of SOMA objects provide:
+
+* `DenseNDArray` a dense, N-dimensional array, with offset (zero-based) integer indexing on each dimension.
```
Comment (ebezzi) [inline README.md:60]:
```suggestion
* `DenseNDArray` is a dense, N-dimensional array, with offset (zero-based) integer indexing on each dimension.
```

File: README.md [hunk 6]
```diff
@@ -7,43 +7,84 @@
...
+* `SparseNDArray` same as `DenseNDArray` but sparse, and it supports point indexing (disjoint index access)
```
Comment (ebezzi) [inline README.md:61]:
```suggestion
* `SparseNDArray` is the same as `DenseNDArray` but sparse, and supports point indexing (disjoint index access)
```

File: README.md [hunk 7]
```diff
@@ -7,43 +7,84 @@
...
+* `DataFrame`  a multi-column table with a user-defined columns names and value types, with support for point indexing.
```
Comment (ebezzi) [inline README.md:62]:
```suggestion
* `DataFrame` is a multi-column table with a user-defined columns names and value types, with support for point indexing. 
```

File: README.md [hunk 8]
```diff
@@ -7,43 +7,84 @@
...
+Reference documentation can also be accessed directly from Python `help(tiledsoma)` or R `help(package = "tiledbsoma")`.
+
+### Main SOMA Objects
+
+The capabilities of TileDB-SOMA lay on the different read, access, and query patterns that each of the main implementations of SOMA objects provide:
+
+* `DenseNDArray` a dense, N-dimensional array, with offset (zero-based) integer indexing on each dimension.
+* `SparseNDArray` same as `DenseNDArray` but sparse, and it supports point indexing (disjoint index access)
+* `DataFrame`  a multi-column table with a user-defined columns names and value types, with support for point indexing. 
+* `Collection` a persistent container of named SOMA objects.
```
Comment (ebezzi) [inline README.md:63]:
```suggestion
* `Collection` is a persistent container of named SOMA objects.
```

**Review by bkmartinjr — COMMENTED** [review]
Suggest adding a section linking back to user's of the package, i.e., "Who is using SOMA", and linking back to Cell Census and other user's who are OK with a public statement of support

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (pablo-gar) [reply README.md:20]: Indeed will update once this is ready

File: README.md [hunk 9]
```diff
@@ -7,43 +7,84 @@
...
+
+## What can TileDB-SOMA do?
+
+Intended to be used for single-cell data, TileDB-SOMA provides Python and R APIs to allow for storage and data access patterns at scale and for larger-than-memory operations:
+
+* Create and write large volumes of data.
+* Open and read data at low latency, locally and from the cloud.
+* Query and access interconnected arrays efficiently and at low latency.
+
+TiledDB-SOMA provides interoperability with existing single-cell toolkits:
```
[the remark this answers is not in the document]
Comment (pablo-gar) [reply README.md:30]: Thanks!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pablo-gar) [reply README.md]: fixed

File: README.md [hunk 10]
```diff
@@ -16,8 +16,8 @@ Embedded](https://github.com/TileDB-Inc/TileDB) to implement the
 Get started on using TileDB-SOMA:
 
 * [Quick start](#quick-start).
-* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma). *Under development*.
-* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma). *Under development*.
+* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma) *Coming soon* .
```
Comment (johnkerl) [inline README.md]: please rebase atop #1103

**Review by johnkerl — CHANGES_REQUESTED** [review]

File: README.md [hunk 11]
```diff
@@ -7,45 +7,92 @@
...
 [SOMA specification](https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md).
 
-# Community
+Get started on using TileDB-SOMA:
 
-* Please join the [TileDB Slack community](https://tiledb-community.slack.com/join/shared_invite/zt-ndq1ipwl-QcithaWG6j1BImtuQGSpag#/shared-invite/email) with dedicated channel `#genomics`
-* Please join the [CZI Slack community](https://cziscience.slack.com/), with dedicated
-channel `#cell-census-users`
+* [Quick start](#quick-start).
+* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma) (*Coming soon*).
```
Comment (johnkerl) [inline README.md:19]:
```suggestion
* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma)
```

this is no longer "coming soon". it is already live.

File: README.md [hunk 12]
```diff
@@ -7,45 +7,92 @@
...
 
-# Community
+Get started on using TileDB-SOMA:
 
-* Please join the [TileDB Slack community](https://tiledb-community.slack.com/join/shared_invite/zt-ndq1ipwl-QcithaWG6j1BImtuQGSpag#/shared-invite/email) with dedicated channel `#genomics`
-* Please join the [CZI Slack community](https://cziscience.slack.com/), with dedicated
-channel `#cell-census-users`
+* [Quick start](#quick-start).
+* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma) (*Coming soon*). 
+* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma) (*Coming soon*).
```
Comment (johnkerl) [inline README.md:20]:
```suggestion
* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma).
```

this is no longer "coming soon". it is already live.

**Review by johnkerl — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -7,45 +7,92 @@
...
 
-# Community
+Get started on using TileDB-SOMA:
 
-* Please join the [TileDB Slack community](https://tiledb-community.slack.com/join/shared_invite/zt-ndq1ipwl-QcithaWG6j1BImtuQGSpag#/shared-invite/email) with dedicated channel `#genomics`
-* Please join the [CZI Slack community](https://cziscience.slack.com/), with dedicated
-channel `#cell-census-users`
+* [Quick start](#quick-start).
+* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma). 
+* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma).
```
Comment (johnkerl) [inline README.md:20]:
```suggestion
* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html). 
* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/).
```

Per our Slack sync :)

**Review by johnkerl — CHANGES_REQUESTED** [review]

**Review by johnkerl — APPROVED** [review]

**Pull request merged**